ts/features/bonus/cgn/components/merchants/CgnMerchantsFilters.tsx

```diff
@@ -0,0 +1,201 @@
+import * as React from "react";
+import { useState } from "react";
+import { Body, Container, Input, Item, Left, Right, View } from "native-base";
+import {
+  FlatList,
+  Keyboard,
+  ListRenderItemInfo,
+  SafeAreaView,
+  ScrollView
+} from "react-native";
+import AppHeader from "../../../../../components/ui/AppHeader";
+import ButtonDefaultOpacity from "../../../../../components/ButtonDefaultOpacity";
+import IconFont from "../../../../../components/ui/IconFont";
+import { H5 } from "../../../../../components/core/typography/H5";
+import { IOColors } from "../../../../../components/core/variables/IOColors";
+import I18n from "../../../../../i18n";
+import { IOStyles } from "../../../../../components/core/variables/IOStyles";
+import { H2 } from "../../../../../components/core/typography/H2";
+import { categories, Category, orders, OrderType } from "../../utils/filters";
+import ItemSeparatorComponent from "../../../../../components/ItemSeparatorComponent";
+import FooterWithButtons from "../../../../../components/ui/FooterWithButtons";
+import {
+  cancelButtonProps,
+  confirmButtonProps
+} from "../../../bonusVacanze/components/buttons/ButtonConfigurations";
+import CategoryCheckbox from "./search/CategoryCheckbox";
+import OrderOption from "./search/OrderOption";
+import { DistanceSlider } from "./search/DistanceSlider";
+
+type Props = {
+  onClose: () => void;
+  onConfirm: () => void;
+  isLocal: boolean;
+};
+
+const CgnMerchantsFilters: React.FunctionComponent<Props> = (props: Props) => {
+  const [searchValue, setSearchValue] = useState<string>("");
+  const [address, setAddress] = useState<string>("");
+  const [checkedCategories, setCheckedCategories] = useState<
+    ReadonlyArray<string>
+  >([]);
+
+  const [selectedOrder, setSelectedOrder] = useState<string>(orders[0].value);
+
+  const [distance, setDistance] = useState<number>(20);
+
+  const onCheckBoxPress = (value: string) => {
+    if (checkedCategories.indexOf(value) !== -1) {
+      setCheckedCategories(checkedCategories.filter(c => c !== value));
+    } else {
+      setCheckedCategories([...checkedCategories, value]);
+    }
+    Keyboard.dismiss();
```

Comment:
```suggestion
    const filteredCategories = checkedCategories.filter(c => c !== value);
    setCheckedCategories(
      checkedCategories.length !== filteredCategories.length
        ? filteredCategories
        : [...checkedCategories, value]
    );
    Keyboard.dismiss();
```
What do you think about this refactoring?
The current implementation iterates checkedCategories twice (`indexOf` + `filter`), the proposed one only 1 time